File: interviewbit-hashing-anagrams.py
class Solution:
    # @param A : tuple of strings
    # @return a list of list of integers
    def anagrams(self, A):
        words = {}
        for i, word in enumerate(A):
            sword = "".join(sorted(list(word)))
            words[sword] = words.get(sword, []) + [i + 1]

        return sorted(words[k] for k in words)

File: interviewbit-hashing-anagrams.java
public class Solution {
    public ArrayList<ArrayList<Integer>> anagrams(final List<String> a) {
        HashMap<String, ArrayList<Integer>> words = new HashMap<String, ArrayList<Integer>>();
        ArrayList<ArrayList<Integer>> indicesList = new ArrayList<ArrayList<Integer>>();
        char[] letters;
        String word;
        for (int i = 0; i < a.size(); i++) {
            letters = a.get(i).toCharArray();
            Arrays.sort(letters);
            word = new String(letters);
            if (!words.containsKey(word)) {
                words.put(word, new ArrayList<Integer>());
            }
            words.get(word).add(i + 1);
        }
        for (String key : words.keySet()) {
            indicesList.add(words.get(key));
        }
        return indicesList;
    }
}
